pkg/signal/signal_unix.go

```diff
@@ -91,10 +92,15 @@ var SignalMap = map[string]syscall.Signal{
 
 // CatchAll catches all signals and relays them to the specified channel.
 func CatchAll(sigc chan os.Signal) {
-	panic("Unsupported on non-linux platforms")
+	handledSigs := make([]os.Signal, 0, len(SignalMap))
+	for _, s := range SignalMap {
+		handledSigs = append(handledSigs, s)
+	}
+	signal.Notify(sigc, handledSigs...)
```

Comment:
Maybe I am missing something but this only uses go standard library functionality so it should work on all platforms.
I think moving this into signal_common.go makes more sense.

Reply:
Good point - I'll move it